lib/internal/Magento/Framework/View/Model/Layout/Merge.php

```diff
@@ -129,6 +130,13 @@ class Merge implements \Magento\Framework\View\Layout\ProcessorInterface
     protected $cacheSuffix;
 
     /**
+     * Cache keys to be able to generate different cache id for same handles
+     *
+     * @var array
+     */
+    protected $cacheKeys = [];
```

Comment:
Currently Magento Technical Guidelines do not encourage designing classes for inheritance. Would be great to make this property private.